rllib/evaluation/sampler.py

```diff
@@ -558,15 +574,21 @@ def _do_policy_eval(tf_sess, to_eval, policies, active_episodes):
         policy = _get_or_raise(policies, policy_id)
         if builder and (policy.compute_actions.__code__ is
                         TFPolicy.compute_actions.__code__):
-            rnn_in_cols = _to_column_format(rnn_in)
+
+            obs_batch = [t.obs for t in eval_data]
+            state_batches = _to_column_format(rnn_in)
+
+            # Call the exploration before_compute_actions hook.
+            policy.exploration.before_compute_actions(
+                obs_batch=obs_batch,
+                timestep=policy.global_timestep,
+                tf_sess=policy.get_session())
```

Comment:
Now that you mentioned we need to call before_compute_actions() in the policy anyways, it makes sense to do it in just the policy classes and never the sampler. Sorry about that, can you revert this change?

Reply:
Done